### ISSUE TITLE
Test-TargetResource SPFarm Resource fails to find local SharePoint Server

### ISSUE THREAD

**Issue by PowershellNinja** [PowerShell/SharePointDsc#795]
**Details of the scenario you try and problem that is occurring:**
I am doing a complete setup of a lab environment for SharePoint, including the setup of Active Directory, SQL and SharePoint on one single vm. The configuration is done using Composite Resources and works fine - until the moment where the farm was already created and you rerun the configuration. As soon as DSC reaches the step
```Powershell
VERBOSE: [SHAREPOINTLAB]:[[SPFarm]CreateSPFarm::[LabSharePoint]SPSetup] Leaving BeginProcessing Method of Get-SPServer.
```
it fails with the error "The server named 'SHAREPOINTLAB' does not exist." (The local server FQDN is SharePointLab.demo.local). The operational eventlog holds the following entry:

_This event indicates that a non-terminating error was thrown when DSCEngine was executing Test-TargetResource on MSFT_SPFarm DSC resource. FullyQualifiedErrorId is Microsoft.SharePoint.PowerShell.SPCmdletGetServer. Error Message is The server named 'SHAREPOINTLAB' does not exist.._

When I just load the SharePoint PSSnapin myself and run Get-SPServer, I get the following result:
```Powershell
Address                        Role                           Status
-------                        ----                           ------
SharepointLab.demo.local       SingleServerFarm               Online
```
I am not sure what is wrong here, the only thing I can imagine is that SharepointLab != SharepointLab.demo.local ?

**The DSC configuration that is using the resource:**
```Powershell
SPFarm CreateSPFarm
{
    Ensure = "Present"
    DatabaseServer = "SharePointLab.demo.local"
    FarmConfigDatabaseName = "SP_Config"
    Passphrase = $FarmPassPhrase
    FarmAccount = $FarmAccount
    PsDscRunAsCredential = $InstallAccount
    AdminContentDatabaseName = "SP_AdminContent"			
    RunCentralAdmin = $true
    DependsOn = "[SPInstall]InstallSharePoint"

}
```
**Version of the Operating System and PowerShell the DSC Target Node is running:**
OSVersion: Windows Server 2016 Version 1607 (OS Build 14393.2155)
PSVersion: 5.1.14393.2155

**Version of SharePoint that is used (e.g. SharePoint 2016):**
SharePoint 2016

**Version of the DSC module you're using:**
Version: 2.2.0.0

**Comment by ykuijs** [comment]
I have reviewed the code and see where this is coming from. I will fix the issue in my next bugfix PR.

**Comment by ykuijs** [comment]
Pretty much, with the exception that I don't use the User DNS domain. It is possible that the server is in a resource domain and the users in a user domain. In that case the User DNS domain is pointing to the incorrect domain ;-)

So I request the computer domain and use that.

**Comment by ykuijs** [comment]
Fixed in PR #811

**Comment by ykuijs** [comment]
PR merged, closing issue
